### ISSUE TITLE
converting .pt to onnx with dynamic flag

### ISSUE THREAD

**Issue by git-hamza** [ultralytics/yolov5#8377]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to convert .pt model to onnx, it successfully gets converted to onnx without the --dynamic flag but with it, it give me some error. I have tried to debug it but haven't had a luck for now. Have anybody else faced the same issue?

### Additional

Execution command followed:
`python3 export.py --weights checkpoints/yolov5n.pt --device 0 --include onnx`

Error:
frame #55: __libc_start_main + 0xf3 (0x7f6a3025c083 in /lib/x86_64-linux-gnu/libc.so.6)
 (function ComputeConstantFolding)
ONNX: export failure: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper__index_select)

Note: Now I know it says that two device are detected but I clearly mentioned device=0 in argument and it works without dynamic flag.

**Comment by glenn-jocher** [comment]
@git-hamza thanks for the bug report! I am able to reproduce an error with the following command. Will ad a TODO to investigate:
```bash
python3 export.py --weights yolov5n.pt  --include onnx --dynamic --device 0
```

**Comment by glenn-jocher** [comment]
@git-hamza needs further investigation but workaround may be just to export on cpu and then run inference on any device:

<img width="1025" alt="Screen Shot 2022-06-28 at 2 14 13 PM" src="https://user-images.githubusercontent.com/26833433/176175924-23f764ac-7796-4ca5-8560-2daae82047a6.png">

**Comment by git-hamza** [comment]
@glenn-jocher Perfect. I will try to do that for now. Thank you for such a rapid reply. Meanwhile If I find the exact culprit, I will update here.

**Comment by glenn-jocher** [comment]
@git-hamza good news 😃! Your original issue may now be fixed ✅ in PR #8378. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
